src/configure/clients/github/TemplateServiceClient.ts

```diff
@@ -21,6 +21,36 @@ export class TemplateServiceClient implements ITemplateServiceClient {
         this.headers = headers;
     }
 
+    public async getTemplatesInfoByFilter(resourceType: string): Promise<TemplateInfo[]> {
```

Comment:
this function should take language, build Target and deployTarget as parameters. The switch case needs to move to the caller

Reply:
The reason is that the client class should be as thin as possible and very specific to constructs of the service it is trying to  call. VS Code business constructs should not be much visible here.

Reply:
cool, will update